### ISSUE TITLE
Using a subdirectory named 'tool' in test makes vscode freeze 

### ISSUE THREAD

**Issue by giggitygoat** [Dart-Code/Dart-Code#4809]
**Describe the bug**
Currently if you make a simple dummy test in a folder named _tool_ it makes vscode freeze and unable to complete. 

**To Reproduce**
1. Create new flutter package project
2. Create folder named _tool_ under _test_
3. Create new file named my_test.dart and insert:
```dart
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('should expect null', () {
    expect(null, isNull);
  });
}
```
4. Click the green arrow to run the test

**Expected behavior**
To execute the file as if it was in the root of the _test_ folder.

**Please complete the following information:**

Dart Code extension: 3.74.0
Flutter extension: 3.74.0 (activated)

App: Visual Studio Code
App Host: desktop
Version: linux 1.82.3

**Comment by DanTup** [comment]
The reason for the error I'm seeing is that when we see something inside a "tool" folder we assume it's Dart (not Flutter), because it's common to put utility scripts inside a Flutter project inside a `tool` folder (which should run with the Dart CLI).

It shouldn't be mis-categorising these inside a nested test/tool folder though, I'll work on fixing that.

**Comment by giggitygoat** [comment]
> I don't see VS Code freeze here, but I see this error in the debug console:
> 
> ![image](https://user-images.githubusercontent.com/1078012/277367634-83ba09ba-381f-45bb-bcd3-642287713e81.png)
> 
> Can you confirm exactly the behaviour you're seeing (what specifically you mean by freeze)?
> 
> (I'll look into what's causing this error in the meantime)

It should be noted my pubspec.yaml looks like this:

```
dev_dependencies:
  flutter_test:
    sdk: flutter
  test: any
```
Which removed the error you are seeing. Furthermore, the freeze I am referring to is that when the test is executing I need to click the stop button some times before it reacts.

**Comment by DanTup** [comment]
> It should be noted my pubspec.yaml looks like this:

Ah, having `test:` in your pubspec is why you're not seeing the same error. If I add that I see the same behaviour as you (and running `dart test test/tool/foo_test.dart` in that case spews a huge number of errors).

The fix I'm working on for the error I posted should solve both issues (by using `flutter` to run the tests). Generally I don't think you should have/need `test` in your pubspec for Flutter tests.

**Comment by giggitygoat** [comment]
> Ah, having `test:` in your pubspec is why you're not seeing the same error. If I add that I see the same behaviour as you (and running `dart test test/tool/foo_test.dart` in that case spews a huge number of errors).

I see, thanks for clarifying!

**Comment by DanTup** [comment]
Fixed by 74fd1ff0
